server/query.go

```diff
@@ -0,0 +1,127 @@
+package server
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tm "github.com/tendermint/tendermint/proto/tendermint/types"
+	coretypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
```

Comment:
So these do not belong here. CLI and public functions, which the CLI uses, for direct Tendermint proxy ABCI methods belongs in `client/rpc`. Put all the CLI methods and the public functions there.